tensorizer/serialization.py

```diff
@@ -54,9 +55,11 @@ class TensorType(Enum):
     STATE_DICT = 2
 
 
-TENSORIZER_VERSION = 1
```

Comment:
@Eta0 - Bringing the tensorizer data version convo here.

It was proposed that we still write a tensorizer data version of `1` that don't use opaque data types since those files will be backwards compatible. All files created with opaque data types will have a data version of `2` since those are not backwards compatible and needs to new code in this PR.

So to implement this we would start with writing the tensorizer version as 1 in the file. If we encounter a tensor that requires an opaque datatype, then we go back and bump the version to 2.

Next time we make a change to the data format we could get rid of this extra versioning logic and bump everything to 3 (or everything to 2). 

The counter would be that it adds too much complexity to be worth it. Bumping the data version to 2 wouldn't make any previously serialized files unsupported. It would only impact the deserialization process if you are using an older tensorizer version on a newly tensorized file.

Is this a correct summary of everything that was discussed in slack?

Reply:
Yes, that is correct. In my opinion a breaking change like that should incur bumping tensorizer to v2.0.0 if it unconditionally produces files incompatible with previous versions, and adding backwards compatibility here would be simple enough in order to not need to do that.